src/util/file_util.cpp

```diff
@@ -61,11 +61,12 @@ std::string get_current_working_directory()
   std::string working_directory=wd;
   free(wd);
   #else
-  char buffer[4096];
+  wchar_t buffer[4096];
```

Comment:
This depends on whether you compile with `-DUNICODE` or not. Since this ever got committed / builds in CI in this state, it looks like some builds do not define `UNICODE`. `TCHAR buffer[4096]` will give a Unicode-agnostic buffer of either 8- or 16-bit chars as per configuration, and similarly some sort of `#ifdef UNICODE` should be used to guard the use of `wstring` below.

Reply:
Oh, thank you, that explains why building via Visual Studio/msbuild yields different results than via `make`.

Reply:
Alternatively of course we could assert that no right-minded person would ever build without `-DUNICODE` any longer and do this everywhere.

Reply:
see https://docs.microsoft.com/en-us/windows/desktop/learnwin32/working-with-strings

Reply:
Ah, that is very useful, thank you!